crates/rover-graphql/src/lib.rs

```diff
@@ -187,9 +236,25 @@ where
                         status_code: resp.status(),
                     }
                 })?;
+
             if let Some(errors) = graphql_response.errors {
                 match graphql_response.data {
-                    Some(data) => Err(GraphQLServiceError::PartialError { data, errors }),
+                    Some(data) => {
+                        let friendly_errors_detail = SimplifiedErrorList::from(&errors).errors;
+
+                        if friendly_errors_detail
+                            .join("")
+                            .contains("Invalid credentials")
+                        {
+                            Err(GraphQLServiceError::InvalidCredentials {})
```

Comment:
@jonathanrainer Thanks for the suggestion.

 What I've got here feels like a nice cleanup in this area of the code, and moving the bulk of the document processing to the the From trait feels nicer. However, I didn't have much luck pushing more of the work/logic to serde. I wonder if you have any specific nudges, or an example of parsing that is similar to what you think this might be improved to?